File: cc/010.cc
// Project Euler Problem 010
//
// Calculate the sum of all the primes below two million.

#include <cstdio>
#include "primes1.h"

int main(int argc, const char* argv[]) {
  Primes1<long> primes;
  long sum = 0;
  long p;
  while ((p = primes.next()) < 2e6) {
    sum += p;
  }
  printf("%ld\n", sum);
}

File: go/010.go
// Project Euler Problem 010
//
// Calculate the sum of all the primes below two million.

package main

import (
	"fmt"

	"./primes"
)

func main() {
	var sum int64 = 0
	i := 0
	for {
		p := primes.Get(i)
		i++
		if p >= 2000000 {
			break
		}
		sum += p
	}
	fmt.Println(sum)
}
